raiden_contracts/deploy/contract_deployer.py

```diff
@@ -211,6 +212,56 @@ def _deploy_and_remember(
             address=deployed_contracts['contracts'][contract_name]['address'],
         )
 
+    def register_token_network(
+            self,
+            token_registry_abi: Dict,
+            token_registry_address: str,
+            token_address: str,
+            channel_participant_deposit_limit: Optional[int],
+            token_network_deposit_limit: Optional[int],
+    ):
+        """Register token with a TokenNetworkRegistry contract."""
+        with_limits = contracts_version_expects_deposit_limits(self.contracts_version)
+        if with_limits:
+            assert channel_participant_deposit_limit is not None, \
```

Comment:
Are you aware that `assert` statements are ignored if python is run with optimizations? It's likely not going to be an issue in most scenarios, but I thought i'd better point it out in case these assertions are required in any case, not just when the `__debug__` variable is truthy.

Reply:
No, I was not aware of this.  This will turn into `if ...: raise ValueError(...)` in this PR.

Reply:
I think the whole codebase needs this kind of cleanup, so I created an issue: https://github.com/raiden-network/raiden-contracts/issues/847